warehouse/ingest-core/src/main/java/datawave/ingest/config/TableConfigCache.java

```diff
@@ -1,64 +1,116 @@
 import java.io.PrintStream;
+import java.util.HashMap;
 import java.util.Map;
 
 public class TableConfigCache extends BaseHdfsFileCacheUtil {
     
-    protected static final Configuration config = new Configuration();
-    
     public static final String ACCUMULO_CONFIG_CACHE_PATH_PROPERTY = "accumulo.config.cache.path";
     public static final String DEFAULT_ACCUMULO_CONFIG_CACHE_PATH = "/data/accumuloConfigCache/accConfCache.txt";
-    public static final String ACCUMULO_CONFIG_CACHE_ENABLE_PROPERTY = "accumulo.config.cache.enable";
+    public static final String ACCUMULO_CONFIG_FILE_CACHE_ENABLE_PROPERTY = "accumulo.config.cache.enable";
     
     protected static final String DELIMITER = "\t";
-    protected static Map<String,String> configMap;
+    private Map<String,Map<String,String>> configMap = new HashMap<>();
+    private static TableConfigCache cache;
     
     protected static final Logger log = Logger.getLogger("datawave.ingest");
     
-    public TableConfigCache(Configuration conf) {
+    private TableConfigCache(Configuration conf) {
         super(conf);
     }
     
-    @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
-    public void setTableConfigs(Configuration conf) {
-        try {
-            TableConfigurationUtil tableConfig = new TableConfigurationUtil(conf);
-            configMap = tableConfig.getTableAggregatorConfigs();
-        } catch (Exception e) {
-            log.error("Unable to get table configurations " + e.getMessage());
+    public static TableConfigCache getCurrentCache(Configuration conf) {
+        if (null == cache) {
+            cache = new TableConfigCache(conf);
         }
+        return cache;
     }
     
-    @Override
-    public void update() {
-        setTableConfigs(this.conf);
-        super.update();
+    public void setTableConfigs(Map<String,Map<String,String>> confMap) {
+        configMap = confMap;
+    }
+    
+    public void clear() {
+        configMap = new HashMap();
+    }
+    
+    public boolean isInitialized() {
+        return !configMap.isEmpty();
     }
     
     @Override
     public void writeCacheFile(FileSystem fs, Path tmpCacheFile) {
         try (PrintStream out = new PrintStream(new BufferedOutputStream(fs.create(tmpCacheFile)), false, "UTF-8")) {
-            for (Map.Entry e : configMap.entrySet()) {
-                out.println(e.getKey() + DELIMITER + e.getValue());
+            for (Map.Entry<String,Map<String,String>> table : configMap.entrySet()) {
+                for (Map.Entry tableProp : table.getValue().entrySet()) {
+                    out.println(table.getKey() + DELIMITER + tableProp.getKey() + DELIMITER + tableProp.getValue());
+                }
             }
         } catch (IOException e) {
             log.error("Unable to write cache file " + tmpCacheFile, e);
         }
     }
     
+    @Override
+    protected void readCache(BufferedReader in, String delimiter) throws IOException {
```

Comment:
curious: why pass in the delimiter here if it is hardcoded in the writeCacheFile routine?